tests/system/providers/google/cloud/ai_platform/example_mlengine.py

```diff
@@ -0,0 +1,381 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG for Google ML Engine service.
+"""
+import os
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict
+
+from airflow import models
+from airflow.models.baseoperator import chain
+from airflow.operators.bash import BashOperator
+from airflow.providers.google.cloud.operators.gcs import GCSCreateBucketOperator, GCSDeleteBucketOperator
+from airflow.providers.google.cloud.operators.mlengine import (
+    MLEngineCreateModelOperator,
+    MLEngineCreateVersionOperator,
+    MLEngineDeleteModelOperator,
+    MLEngineDeleteVersionOperator,
+    MLEngineGetModelOperator,
+    MLEngineListVersionsOperator,
+    MLEngineSetDefaultVersionOperator,
+    MLEngineStartBatchPredictionJobOperator,
+    MLEngineStartTrainingJobOperator,
+    MLEngineTrainingCancelJobOperator,
+)
+from airflow.providers.google.cloud.transfers.local_to_gcs import LocalFilesystemToGCSOperator
+from airflow.providers.google.cloud.utils import mlengine_operator_utils
+from airflow.utils.trigger_rule import TriggerRule
+
+PROJECT_ID = os.environ.get("GCP_PROJECT_ID")
+ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID")
+
+DAG_ID = "example_gcp_mlengine"
+
+BUCKET_NAME = f"bucket-{DAG_ID}-{ENV_ID}"
+REGION = "us-central1"
+
+MODEL_NAME = "model_name_test"
+SAVED_MODEL_PATH = f"gs://{BUCKET_NAME}/keras-job-dir/keras_export/"
+JOB_DIR = f"gs://{BUCKET_NAME}/keras-job-dir"
+JOB_NAME = "training-job-keras"
+JOB_ID = "training-job-test"
+
+PREDICTION_INPUT = f"gs://{BUCKET_NAME}/prediction_input.json"
+PREDICTION_OUTPUT = f"gs://{BUCKET_NAME}/prediction_output"
+
+TRAINER_URI = f"gs://{BUCKET_NAME}/trainer-0.1.tar.gz"
+TRAINER_PY_MODULE = "trainer.task"
+
+SUMMARY_TMP = f"gs://{BUCKET_NAME}/tmp/"
+SUMMARY_STAGING = f"gs://{BUCKET_NAME}/staging/"
+
+FILE_NAME = "trainer-0.1.tar.gz"
+FILE_NAME_1 = "prediction_input.json"
+
+CURRENT_FOLDER = Path(__file__).parent
+FILE_LOCAL_PATH = str(Path(CURRENT_FOLDER) / "resources" / f"{FILE_NAME}")
+FILE_LOCAL_PATH_1 = str(Path(CURRENT_FOLDER) / "resources" / f"{FILE_NAME_1}")
+
+
+hyperparams: Dict[str, Any] = {
+    'goal': 'MAXIMIZE',
+    'hyperparameterMetricTag': 'metric1',
+    'maxTrials': 30,
+    'maxParallelTrials': 1,
+    'enableTrialEarlyStopping': True,
+    'params': [],
+}
+
+hyperparams['params'].append(
+    {
+        'parameterName': 'hidden1',
+        'type': 'INTEGER',
+        'minValue': 40,
+        'maxValue': 400,
+        'scaleType': 'UNIT_LINEAR_SCALE',
+    }
+)
+
+hyperparams['params'].append(
+    {'parameterName': 'numRnnCells', 'type': 'DISCRETE', 'discreteValues': [1, 2, 3, 4]}
+)
```

Comment:
Why not create the list literal instead of appends afterwards, ie:

```
hyperparams: Dict[str, Any] = {
    'goal': 'MAXIMIZE',
    'hyperparameterMetricTag': 'metric1',
    'maxTrials': 30,
    'maxParallelTrials': 1,
    'enableTrialEarlyStopping': True,
    'params': [
    {
        'parameterName': 'hidden1',
        'type': 'INTEGER',
        'minValue': 40,
        'maxValue': 400,
        'scaleType': 'UNIT_LINEAR_SCALE',
    },
   ... other items
],
}
```